Sources/tbDEX/crypto/JWK.swift

```diff
@@ -0,0 +1,148 @@
+        case encryption = "enc"
+    }
+
+    /// Supported `key_ops` values.
+    public enum KeyOperations: String, Codable {
+        case encrypt
+        case decrypt
+        case sign
+        case verify
+        case deriveKey
+        case driveBits
+        case wrapKey
+        case unwrapKey
+    }
+
+    // MARK: - Common JWK Properties
+
+    /// The below properties represent JWK parameters that are common amongst multiple JWK key types.
+    /// See [JSON Web Key (JWK) Format](https://datatracker.ietf.org/doc/html/rfc7517#section-4) for more information
+    /// on any of these parameters.
+
+    /// The `kty` (key type) parameter identifies the cyrptographic algorithm family used with the key.
+    public var keyType: KeyType
+
+    /// The `use` (public key use) parameter identifies the intended use of the public key.
+    public var publicKeyUse: PublicKeyUse?
+
+    /// The "key_ops" (key operations) parameter identifies the operation(s) for which the key is intended to be used.
+    public var keyOperations: [KeyOperations]?
+
+    /// The `alg` (algorithm) parameter identifies the cryptographic algorithm intended for use with the key.
+    public var algorithm: Algorithm?
+
+    /// The "alg" (algorithm) parameter identifies the algorithm intended for use with the key.
+    public var keyIdentifier: String?
+
+    /// The `crv` (curve) parameter identifies the cryptographic curve intended for use with the key.
+    public var curve: Curve?
+
+    /// The "x5u" (X.509 URL) parameter is a URI [RFC3986](https://datatracker.ietf.org/doc/html/rfc3986) that refers
+    /// to a resource for an X.509 public key certificate or certificate chain
+    /// [RFC5280](https://datatracker.ietf.org/doc/html/rfc5280).
+    public var x509Url: String?
+
+    /// The "x5c" (X.509 certificate chain) parameter contains a chain of one or more PKIX certificates
+    /// [RFC5280](https://datatracker.ietf.org/doc/html/rfc5280)
+    public var x509CertificateChain: String?
+
+    /// The "x5t" (X.509 certificate SHA-1 thumbprint) parameter is a base64url-encoded SHA-1 thumbprint (a.k.a. digest)
+    /// of the DER encoding of an X.509 certificate [RFC5280](https://datatracker.ietf.org/doc/html/rfc5280).
+    public var x509CertificateSha1Thumbprint: String?
+
+    /// The "x5t#S256" (X.509 certificate SHA-256 thumbprint) parameter is a base64url-encoded SHA-256 thumbprint
+    /// (a.k.a. digest) of the DER encoding of an X.509 certificate
+    /// [RFC5280](https://datatracker.ietf.org/doc/html/rfc5280).
+    public var x509CertificateSha256Thumbprint: String?
+
+    // MARK: - KeyType Specific JWK Properties
+
+    /// The below properties represent JWK parameters that are unique to specific JWK key types.
+
+    /// `d` Private exponent.
+    public var d: String?
+
+    /// The x-coordinate for the Elliptic Curve point.
+    public var x: String?
+
+    /// Elliptic Curve y-coordinate.
+    public var y: String?
```

Comment:
is there a JWK lib we could use instead here? there are other JWK properties that might show up

maybe https://github.com/airsidemobile/JOSESwift

Reply:
I opted not to do this, as our usage for JWKs in tbDEX is very limited, and these libraries are much larger than we actually need. On top of that, they've modeled certain aspects of their implementation with enums, which we cannot extend as they are closed types. 

For these reasons, I've opted to make a super minimal representation of JWK for now. We can always revisit in the future.